src/routers/alpha-router/alpha-router.ts

```diff
@@ -1631,6 +1642,86 @@ export class AlphaRouter
     }
 
     if (!swapRouteRaw) {
+      // this is a well-intended scoped fix,
+      // because we already know that in all the reported cached routes invalid case,
+      // end user cannot get quote.
+      // therefore its safe to limit the cached routes cache invalidation when there;s no quote.
+      if (
+        Math.random() * 100 <
+        (this.cachedRoutesCacheInvalidationFixRolloutPercentage ?? 0)
+      ) {
+        // there's a possibility that cachedRoutes object is not null/undefined, but it could cause swapRouteFromCache to be null/undefined
+        // because such cached routes already becomes invalid.
+        // in that case then, we want to try swapRouteFromChain.
+        // If CachedRoutes is populated, then swapRouteFromChain is not being invoked at all in
+        // https://github.com/Uniswap/smart-order-router/blob/66927c4e4f7e75138df48afab1c33a6c6d8d8f15/src/routers/alpha-router/alpha-router.ts#L1469
+        if (
+          this.routeCachingProvider &&
+          routingConfig.writeToCachedRoutes &&
```

Comment:
even if `writeToCachedRoutes` is false, dont we want to do `getSwapRouteFromChain` if`swapRouteFromCache` is invalid/null?

Reply:
i guess by definition no, but `writeToCachedRoutes` is nowadays always set to true right?